__tests__/app.test.js

```diff
@@ -158,3 +166,65 @@ describe("GET /api/reviews/:review_id/comments", () => {
       });
   });
 });
+describe("POST /api/reviews/:review_id/comments", () => {
+  test("should create a new comment and respond with an object of posted comment and other info", () => {
+    const review_id = 3;
+    const requestBody = {
+      username: "mallionaire",
+      body: "Posted successfully!",
+    };
+    return request(app)
+      .post(`/api/reviews/${review_id}/comments`)
+      .send(requestBody)
+      .expect(201)
+      .expect((response) => {
+        const comment = response.body.comment;
+        expect(comment.review_id).toBe(review_id);
+        expect(comment.author).toBe(requestBody.username);
+        expect(comment.body).toBe(requestBody.body);
+        expect(comment).toHaveProperty("comment_id");
+        expect(comment).toHaveProperty("votes");
+        expect(comment).toHaveProperty("created_at");
+      });
+  });
+
+  test("should get 404 if given an ID which does not exist...yet", () => {
+    const reviewId = 1002;
+    return request(app)
+      .post(`/api/reviews/${reviewId}/comments`)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe(`Try again - ID does not exist yet!!!`);
+      });
+  });
+
+  test("should get 404 if given an username does not exist", () => {
+    const reviewId = 3;
+    const requestBody = {
+      username: "not-here",
+      body: "Posted successfully!",
+    };
+    return request(app)
+      .post(`/api/reviews/${reviewId}/comments`)
+      .send(requestBody)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Try again - Username does not exist yet!!!");
+      });
+  });
+
+  test("should get 400 error if given bad path/invalid syntax", () => {
+    const review_id = "pear";
+    const requestBody = {
+      username: "mallionaire",
+      body: "Posted successfully!",
+    };
+    return request(app)
+      .post(`/api/reviews/${review_id}/comments`)
+      .send(requestBody)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("bad request");
+      });
```

Comment:
Please add a 201 for unnecessary properties and a 400 for missing required field eg, no username or body.